sdk/export/metric/metric.go

```diff
@@ -21,70 +21,271 @@ import (
-// to use and gathering exported results from the SDK.  The standard SDK
-// supports binding only one of these interfaces, i.e., a single exporter.
+// Batcher is responsible for deciding which kind of aggregation to
+// use (via AggregationSelector), gathering exported results from the
+// SDK during collection, and deciding over which dimensions to group
+// the exported data.
 //
-// Multiple-exporters could be implemented by implementing this interface
-// for a group of Batcher.
+// The SDK supports binding only one of these interfaces, as it has
+// the sole responsibility of determining which Aggregator to use for
+// each record.
+//
+// The embedded AggregationSelector interface is called (concurrently)
+// in instrumentation context to select the appropriate Aggregator for
+// an instrument.
+//
+// The `Process` method is called during collection in a
+// single-threaded context from the SDK, after the aggregator is
+// checkpointed, allowing the batcher to build the set of metrics
+// currently being exported.
+//
+// The `ReadCheckpoint` method is called during collection in a
+// single-threaded context from the Exporter, giving the exporter
+// access to a producer for iterating over the complete checkpoint.
 type Batcher interface {
-	// AggregatorFor should return the kind of aggregator
-	// suited to the requested export.  Returning `nil`
-	// indicates to ignore the metric update.
+	// AggregationSelector is responsible for selecting the
+	// concrete type of Aggregator used for a metric in the SDK.
+	//
+	// This may be a static decision based on fields of the
+	// Descriptor, or it could use an external configuration
+	// source to customize the treatment of each metric
+	// instrument.
+	//
+	// The result from AggregatorSelector.AggregatorFor should be
+	// the same type for a given Descriptor or else nil.  The same
+	// type should be returned for a given descriptor, because
+	// Aggregators only know how to Merge with their own type.  If
+	// the result is nil, the metric instrument will be disabled.
+	//
+	// Note that the SDK only calls AggregatorFor when new records
+	// require an Aggregator. This does not provide a way to
+	// disable metrics with active records.
+	AggregationSelector
+
+	// Process is called by the SDK once per internal record,
+	// passing the descriptor, the corresponding labels, and the
+	// checkpointed aggregator.  The Batcher should be prepared to
+	// process duplicate (Descriptor, Labels) pairs during this
+	// pass due to race conditions, but this will usually be the
+	// ordinary course of events, as Aggregators are merged to
+	// reduce their dimensionality (i.e., group-by).
 	//
-	// Note: This is context-free because the handle should not be
-	// bound to the incoming context.  This call should not block.
-	AggregatorFor(Record) Aggregator
+	// The Context argument originates from the controller that
+	// orchestrates collection.
+	Process(ctx context.Context,
```

Comment:
Why did you change the `Record` argument to the three variables of a `Record`?

Reply:
It wasn't intentional. There were two different record-like types before, and the outcome after some refactoring w/ a single export.Record struct feels right. A lot changed (for the better), but I just missed the opportunity to replace these three by the new Record. Fixed.